### PR TITLE
IMG-112: Adding further test coverage for rendering RGB 16 bit images.

### PR DESCRIPTION
**Pull request by brjeter**
- Added multiple test images that represent different rendering scenarios.
- Added unit tests to test the above images.
- Removed some 16-bit specific code from the AbstractRgbImageRepresentationHandler and put it in the Rgb48ImageRepresentationHandler

Reviewers:
@dcruver @bradh

### CODE REVIEW

File: render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java [hunk 1]
```diff
@@ -20,10 +20,23 @@
...
+
 class Rgb48ImageRepresentationHandler extends AbstractRgbImageRepresentationHandler {
+    private final int bitsToDiscard;
+
     Rgb48ImageRepresentationHandler(final Map<Integer, Integer> bandMap,
-            final int actualBitsPerPixelPerBand) {
+            final int actualBitsPerPixelPerBand, final PixelJustification pixelJustification) {
         super(bandMap, actualBitsPerPixelPerBand);
+        if (actualBitsPerPixelPerBand - Byte.SIZE < 0) {
+            this.bitsToDiscard = 0;
```
Comment (bradh) [inline render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java]: The codecov results show we are never hitting this line. (i.e. the test for `actualBitsPerPixelPerBand - Byte.SIZE < 0` is never true). Is that expected? I don't think it is valid in NITF, but if it was (say NBPP== 16, ABPP == 7) then perhaps we may need to shift depending on PixelJustification.

File: render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java [hunk 2]
```diff
@@ -20,10 +20,23 @@
...
+        } else {
+            if (pixelJustification.equals(PixelJustification.RIGHT)) {
+                this.bitsToDiscard = actualBitsPerPixelPerBand - Byte.SIZE;
+            } else {
+                this.bitsToDiscard = Byte.SIZE;
```
Comment (bradh) [inline render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java]: Just to be clear, this case is really that we are 16 bits per pixel, and we want to be 8 bits per pixel. So to extend this to the NBPP=12 or NBPP=32 cases, we are really discarding NBPP - Byte.SIZE?

File: render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java (hunk 1 above)
Comment (brjeter) [inline render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java]: According to MIL-STD-2500A Table III, the ABPP and NBPP can both be between 1-64. Not sure if this conditional actually ever happens, but the specification seems to allow for it. That's a really good point though: if this conditional is met, we'll need to shift based on the pixel justification. I will put in the fix and see if I can write up a unit test that tests this case.

File: render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java (hunk 2 above)
Comment (brjeter) [inline render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java]: Yes that's true.

**Comment by brjeter** [comment]
I have updated the unit tests to cover every line. Let me know if there are any other issues.

**Comment by bradh** [comment]
Thanks very much. Lets do this again some time!

**Comment by brjeter** [comment]
You're welcome! It was good working with you.

**Pull request merged**